make deployment status behaviour more descriptive (#31226)
kubernetes/website: content/en/docs/concepts/workloads/controllers/deployment.md

* The Deployment is scaling down its older ReplicaSet(s).
* New Pods become ready or available (ready for at least [MinReadySeconds](#min-ready-seconds)).

> When the rollout becomes progressing, the Deployment controller adds a DeploymentCondition with the following

Reviewer: ```suggestion
When the rollout becomes “progressing”, the Deployment controller adds a condition with
the following
```
`DeploymentCondition` isn't part of the Kubernetes API; it's an implementation detail of the Golang implementation.
_In theory_ someone could drop in their own conformant Deployment controller and use that. We should write docs that are still relevant for that case.

attributes to the Deployment's `.status.conditions`:

> * Type=Progressing
> * Status=True
> * Reason=NewReplicaSetCreated | FoundNewReplicaSet | ReplicaSetUpdated

Reviewer: Although the existing text uses this style, I'd rather not see more of it. How about switching to something like:

```suggestion
* `type: Progressing`
* `status: "true"`
* `reason: NewReplicaSetCreated` | `reason: FoundNewReplicaSet` | `reason: ReplicaSetUpdated`
```

(I think) this is how they'd look in YAML, which is how we usually represent API details. JSON is also OK.

Reviewer: Does the Deployment controller also emit an Event? If so, maybe we can mention that?

Author: I would rather not in this case. 
1. It emits scaling event. Eg. `Scaled up replica set test to 1`. But this is not reflective of a rollout / deployment and occurs in plain scaling as well.
2. If you create Deployment with 0 replicas there is no event emitted.

* All of the replicas associated with the Deployment are available.
* No old replicas for the Deployment are running.

> When the rollout completes, the Deployment controller sets a DeploymentCondition with the following

Reviewer: see previous comment - let's try to show how things would look in the API.

attributes to the Deployment's `.status.conditions`:

* Type=Progressing
* Status=True

> * Reason=NewReplicaSetAvailable

Reviewer: see previous comment - let's try to show how things would look in the API.

> This condition will stay `True`, until a new rollout is initiated (even when availability of replicas changes, in that case see `Type=Available` condition).

Reviewer: ```suggestion
This `Progressing` condition will retain a status value of `"True"` until a new rollout
is initiated. The condition holds even when availability of replicas changes (which
does instead affect the `Available` condition).
```
?
